website/blog/2024-05-14-monitor-ferretdb-performance-using-coroot.md

```diff
@@ -0,0 +1,180 @@
+---
+slug: monitor-ferretdb-performance-using-coroot
+title: 'Monitor FerretDB Performance using Coroot'
+authors: [alex]
+description: >
+  Observability is vital in any infrastructure. Find out how Coroot can provide real-time monitoring and visibility into a FerretDB setup.
+image: /img/blog/ferretdb-coroot.jpg
+tags: [tutorial, community, cloud, open source]
+---
+
+![Monitor FerretDB Performance using Coroot](/img/blog/ferretdb-coroot.jpg)
+
+Effective real-time monitoring is a critical aspect of any infrastructure.
+[Coroot](https://coroot.com/) is an open source observability platform that can provide real-time monitoring and visibility into a [FerretDB](https://www.ferretdb.com/) setup.
+
+<!--truncate-->
+
+Effective real-time monitoring is a critical aspect of any infrastructure.
+[Coroot](https://coroot.com/) is an open source observability platform that can provide real-time monitoring and visibility into a [FerretDB](https://www.ferretdb.com/) setup.
+App developers can quickly troubleshoot issues without necessarily becoming experts in infrastructure operations and management.
+
+If you're just finding out about FerretDB, it's the truly open source document database that provides MongoDB compatibility with Postgres as the backend.
+With Coroot, you can get visibility into your entire system in a matter of minutes.
+
+In this blog post, you'll learn about Coroot and how it can help you monitor and get better visibility into your FerretDB instance.
+
+## Coroot observability for FerretDB
+
+Setting up effective monitoring for your entire system can be resource-intensive, time-consuming, and expensive.
+Imagine a scenario where your FerretDB instance is deployed as a Docker container along with other components and services.
+Observability into all critical areas of the system in as little time is vital.
+
+With Coroot, you can visualize every aspect of your entire setup.
+That includes all its components, including your applications and their subsystems, databases, and, most importantly, how they all communicate.
+
+Coroot heavily relies on eBPF, a technology that enhances observability and monitoring by eliminating the need to instrument application code manually.
+Custom programs can run directly within the Linux kernel without changing kernel source code or loading additional kernel modules.
+You get deeper and more extensive information on system calls, network functions, and security issues – just better observability.
+
+## Setting up Coroot for FerretDB monitoring
+
+Since Coroot uses eBPF, you need the right environment before setting it up.
+The most recent versions of the Linux kernel (v 4.16 and above) should be compatible since they offer at least minimal eBPF support.
```

Comment:
```suggestion
The most recent versions of the Linux kernel (v4.16 and above) should be compatible since they offer at least minimal eBPF support.
```